flooder/src/main.rs

```diff
@@ -1,100 +1,277 @@
+    utility::BlocksApi,
+    AccountId, Balance, KeyPair, Nonce, SignedConnection, SignedConnectionApi, TxStatus, TOKEN,
 };
 use clap::Parser;
 use config::Config;
 use futures::future::join_all;
-use log::info;
-use subxt::ext::sp_core::{sr25519, Pair};
-use tokio::{time, time::sleep};
+use log::{debug, info};
+use subxt::{
+    config::{extrinsic_params::BaseExtrinsicParamsBuilder, substrate::Era},
+    ext::sp_core::{sr25519, Pair},
+    tx::TxPayload,
+    utils::{MultiAddress, Static},
+};
+use tokio::time::{sleep, Duration, Instant};
 
 mod config;
 
+fn transfer_keep_alive(dest: AccountId, amount: Balance) -> impl TxPayload + Send + Sync {
+    aleph_client::api::tx()
+        .balances()
+        .transfer_keep_alive(MultiAddress::Id(Static(dest)), amount)
+}
+
+fn transfer_all(dest: AccountId, keep_alive: bool) -> impl TxPayload + Send + Sync {
+    aleph_client::api::tx()
+        .balances()
+        .transfer_all(MultiAddress::Id(Static(dest)), keep_alive)
+}
+
+struct Schedule {
+    pub intervals: u64,
+    pub interval_duration: u64,
+    pub transactions_in_interval: u64,
+}
+
 async fn flood(
-    connections: Vec<SignedConnection>,
+    connections: &Vec<SignedConnection>,
     dest: AccountId,
     transfer_amount: Balance,
-    tx_count: u64,
-    rate_limiting: Option<(u64, u64)>,
+    schedule: Schedule,
     status: TxStatus,
-) -> anyhow::Result<()> {
-    let handles: Vec<_> = connections
-        .into_iter()
-        .map(|conn| {
-            let dest = dest.clone();
-            tokio::spawn(async move {
-                let (time, (tx_count, round_count)) = match rate_limiting {
-                    Some((tx_in_interval, interval_secs)) => (
-                        interval_secs,
-                        (tx_in_interval, (tx_count + tx_in_interval) / tx_in_interval),
-                    ),
-                    _ => (0, (tx_count, 1)),
-                };
-
-                for i in 0..round_count {
-                    info!("starting round #{}", i);
-                    let start = time::Instant::now();
-
-                    info!("sending #{} transactions", tx_count);
-                    for _ in 0..tx_count {
-                        conn.transfer_keep_alive(dest.clone(), transfer_amount, status)
-                            .await
-                            .unwrap();
-                    }
+    pool_limit: u64,
+) -> anyhow::Result<Vec<Nonce>> {
+    let start = Instant::now();
+    let start_finalized_hash = connections[0].get_finalized_block_hash().await?;
+    let start_finalized_number = connections[0]
+        .get_block_number(start_finalized_hash)
+        .await?
+        .unwrap() as u64;
+
+    let mut nonces: Vec<u32> = vec![];
+    for conn in connections {
+        nonces.push(conn.account_nonce(conn.account_id()).await?);
+    }
+
+    let mut overdue_transactions = 0;
+    for i in 0..schedule.intervals {
+        let pending_in_pool = connections[0].pending_extrinsics_len().await?;
```

Comment:
I'd rather stay with only option 2, as unfortunately for 1) it seems, that:
* when submitting a transaction that exceeds tx-pool's pool limit, it will be accepted, but some other transaction (with lower priority, or the oldest one in case of draws) will be dropped instead, with all transactions depending on it, i.e. with the same sender, but higher nonce. This will make deciding which transaction are dropped kida unpredictable, which complicates nonce computation.
* When node's txpool is almost full, it starts behaving strangely when I test it - maybe rpc node doesn't propagate all transactions to other nodes in the network, idk and haven't had time to debug.